scripts/main/contextMenu.js

```diff
@@ -474,23 +524,74 @@ contextMenu.photoMore = function (photoID, e) {
 };
 
+/**
+ * @param {Album[]} albums
+ * @param {string} albumID
+ *
+ * @returns {string[]}
+ */
 contextMenu.getSubIDs = function (albums, albumID) {
-	let ids = [parseInt(albumID, 10)];
-	let a;
+	let ids = [albumID];
 
-	for (a = 0; a < albums.length; a++) {
-		if (parseInt(albums[a].parent_id, 10) === parseInt(albumID, 10)) {
-			ids = ids.concat(contextMenu.getSubIDs(albums, albums[a].id));
+	albums.forEach(function (album) {
+		if (album.parent_id === albumID) {
+			ids = ids.concat(contextMenu.getSubIDs(albums, album.id));
 		}
 
-		if (albums[a].albums && albums[a].albums.length > 0) {
-			ids = ids.concat(contextMenu.getSubIDs(albums[a].albums, albumID));
+		if (album.albums && album.albums.length > 0) {
+			ids = ids.concat(contextMenu.getSubIDs(album.albums, albumID));
 		}
-	}
+	});
 
 	return ids;
 };
 
+/**
+ * @callback TargetAlbumSelectedCB
+ *
+ * Called by {@link contextMenu.move} after the user has selected a target ID.
+ * In most cases, {@link album.setAlbum} or {@link photo.setAlbum} are
+ * directly used as the callback.
+ * This design decision is the only reason, why this callback gets more
+ * parameters than the selected target ID.
+ * The parameter signature of this callback matches {@link album.setAlbum}.
+ *
+ * However, the callback should actually enclose all other parameters it
+ * needs and only receive the target ID.
+ *
+ * TODO: Re-factor callbacks.
+ *
+ * @param {string[]} IDs      the source IDs
+ * @param {?string} targetID  the ID of the target album
+ * @param {boolean} [confirm] indicates whether the callback shall show a
+ *                            confirmation dialog to the user for whatever to
+ *                            callback is going to do
+ * @returns {void}
+ */
+
+/**
+ * Shows the context menu with the album tree and allows the user to select a target album.
+ *
+ * **ATTENTION:** The name `move` of this method is very badly chosen.
+ * The method does not move anything, but only shows the menu and reports
+ * the selected album.
+ * In particular, the method is used by any operation which needs an target
+ * album (i.e. merge, copy-to, etc.)
+ *
+ * TODO: Find a better name for this function.
+ *
+ * The method calls the provided callback after the user has selected a
+ * target album and passes the ID of the target album together with the
+ * source `IDs` and the event `e` to the callback.
+ *
+ * TODO: Actually the callbacks should enclose all additional parameters (e.g., `IDs`) they need. Refactor the callbacks.
+ *
+ * @param {string[]} IDs - IDs of source objects (either album or photo IDs)
+ * @param {jQuery.Event} e - Some (?) event
+ * @param {TargetAlbumSelectedCB} callback - to be called after the user has selected a target ID
+ * @param {string} [kind=UNSORTED] - Name of root album; either "UNSORTED" or "ROOT"; TODO: Why do we need two different names for the same thing?
+ * @param {boolean} [display_root=true] - Whether the root (aka unsorted) album shall be shown
+ */
```

Comment:
This was because merge, move, duplicate were using pretty much the same code in that part. 
Before `Copy to... ` was implemented server side, it was a combination of two API calls: duplicate followed by move.

Reply:
I don't understand that explanation. Even if "copy to" was defined as "duplicate" followed by "move" in the past, I don't see the connection to the second parameter and why the root folder is sometimes named "unsorted" and sometimes "root".

Reply:
The answer was for why there is a misnamed _move_ function.
As for _root_ and _unsorted_, just duplication... :|